ios/Plugin/Plugin.swift

```diff
@@ -198,6 +200,127 @@ public class NativeBiometric: CAPPlugin {
+                    call.resolve(obj)
+                }
+            }catch{
+                call.reject("Cannot generate public key")
+            }
+        }
+    }
+    
+    @objc func signData(_ call: CAPPluginCall){
+        var obj = JSObject()
+        
+        let query: [String: Any] = [
+            kSecClass as String: kSecClassKey,
+            kSecAttrKeyClass as String: kSecAttrKeyClassPrivate,
+            kSecAttrKeyType as String: kSecAttrKeyTypeRSA,
+            kSecAttrKeySizeInBits as String: 2048,
+            kSecAttrApplicationTag as String: privateKeyTag,
+            kSecReturnRef as String: true
+        ]
+        
+        var item: CFTypeRef?
+        let status = SecItemCopyMatching(query as CFDictionary, &item)
+        guard status == errSecSuccess else { return }
+        let privateKey = item as! SecKey
+        
+        guard let challengeString = call.getString("challengeString") else {
+            call.reject("No challenge string was provided")
+            return
+        }
+        
+        guard let messageData = challengeString.data(using: String.Encoding.utf8) else {
+            call.reject("Invalid message to sign")
+            return
+        }
+        
+        do{
+            guard let signData = SecKeyCreateSignature(
+            privateKey,
+            SecKeyAlgorithm.ecdsaSignatureMessageX962SHA256,
+            messageData as CFData, nil) else {
+                call.reject("Cannot sign data")
+                return
+            }
+            
+            let signedData = signData as Data
+            let signedString = signedData.base64EncodedString(options: [])
+            obj["signedData"] = signedString
+            call.resolve(obj)
+        }
+    }
+    
+    func storePublicKeyToKeychain(_ key: SecKey) throws {
+        let tag = publicKeyTag.data(using: .utf8)!
+        let addquery: [String: Any] = [kSecClass as String: kSecClassKey,
+                                       kSecAttrApplicationTag as String: tag,
+                                       kSecValueRef as String: key]
+        
+        let status = SecItemAdd(addquery as CFDictionary, nil)
+        guard status == errSecSuccess else { return }
+    }
+    
+    func getPublicFromKeychain() throws -> SecKey {
+        let getquery: [String: Any] = [kSecClass as String: kSecClassKey,
+                                       kSecAttrApplicationTag as String: publicKeyTag,
+                                       kSecAttrKeyType as String: kSecAttrKeyTypeRSA,
+                                       kSecReturnRef as String: true]
+        
+        var item: CFTypeRef?
+        let status = SecItemCopyMatching(getquery as CFDictionary, &item)
+        guard status == errSecSuccess else { throw KeychainError.noPassword}
+        let key = item as! SecKey
+        
+        return key
+    }
+    
+    func generatePublicKey() throws -> SecKey {
+        let attributes: [String: Any] = [
```

Comment:
Should we lock the access to only can be accessed when user done biometrics with `kSecAccessControlBiometryAny`?

Ref: https://developer.apple.com/documentation/security/keychain_services/keychain_items/restricting_keychain_item_accessibility

Reply:
I was thinking of having control access of the keychain, however it might trigger double biometric prompt. Shall we implement `kSecAttrAccessibleWhenUnlocked` instead?

Reply:
I see, if it actually trigger double biometric prompt then `kSecAttrAccessibleWhenUnlocked` is enough :ok_hand: